### PR TITLE
Allow sorting through configuration in ListAction

### PR DESCRIPTION
**Pull request by ArnaudBenassy**
Will be usable only after https://github.com/Elao/ElaoAdminBundle/pull/13 merge

Allow to add configuration options to sort entities

### CODE REVIEW

File: Action/ListAction.php [hunk 1]
```diff
@@ -155,10 +155,24 @@ protected function getFilters(Form $form = null)
...
+                    $this->parameters['sorting']['sort_by'] => $this->parameters['sorting']['sort_direction']
+                ] : []
+            );
         }
 
-        return $this->paginate($request, $this->modelManager->getTarget($filters));
+        $target = $this->modelManager->getTarget($filters);
+
+        if ($this->parameters['sorting']['enabled']) {
+            $target->addsortBy(
```
Comment (benji07) [inline Action/ListAction.php]: addSortBy

Comment (benji07) [inline Action/ListAction.php]: what if the $target is an array ?

Comment (ArnaudBenassy) [inline Action/ListAction.php]: Worse... bad bulk renaming : addSortBy did not exist, addOrderBy do 😳

**Comment by ArnaudBenassy** [comment]
Some corrections have been done

File: Action/ListAction.php [hunk 2]
```diff
@@ -155,10 +157,24 @@ protected function getFilters(Form $form = null)
     public function getModels(Request $request, array $filters = [])
     {
         if (!$this->parameters['pagination']['enabled']) {
-            return $this->modelManager->findAll($filters);
+            return $this->modelManager->findAll(
+                $filters,
+                $this->parameters['sorting']['enabled'] ? [
```
Comment (Tom32i) [inline Action/ListAction.php:162]: Seems a bit complex and not very flexible, don't you think?

File: DependencyInjection/ActionConfiguration/ListActionConfiguration.php [hunk 3]
```diff
@@ -38,6 +38,17 @@ protected function buildParametersTree(NodeParentInterface $node)
                     ->end()
                 ->end()
             ->end()
+            ->arrayNode('sorting')
+                ->canBeEnabled()
+                ->children()
+                    ->scalarNode('sort_by')
```
Comment (Tom32i) [inline DependencyInjection/ActionConfiguration/ListActionConfiguration.php:44]: So you can only sort by one field?

Comment (ArnaudBenassy) [inline DependencyInjection/ActionConfiguration/ListActionConfiguration.php:44]: Yep, better than nothing :)

... The real point is : I'm not really comfortable with configuration, and it was sufficient for my needs 😳

**Comment by ArnaudBenassy** [comment]
Nope, better to extend DoctrineModelManager...

**Pull request closed without merging**